aws/resource_aws_redshift_snapshot_schedule_association_test.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/redshift"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSRedshiftSnapshotScheduleAssociation_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	rName := acctest.RandString(8)
+	resourceName := "aws_redshift_snapshot_schedule_association.default"
+	snapshotScheduleResourceName := "aws_redshift_snapshot_schedule.default"
+	clusterResourceName := "aws_redshift_cluster.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRedshiftSnapshotScheduleAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRedshiftSnapshotScheduleAssociationConfig(rInt, rName, "rate(12 hours)"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftSnapshotScheduleAssociationExists(resourceName),
+					resource.TestCheckResourceAttrPair(resourceName, "schedule_identifier", snapshotScheduleResourceName, "id"),
+					resource.TestCheckResourceAttrPair(resourceName, "cluster_identifier", clusterResourceName, "id"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSRedshiftSnapshotScheduleAssociationDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_redshift_snapshot_schedule_association" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).redshiftconn
+		clusterIdentifier, scheduleIdentifier, err := resourceAwsRedshiftSnapshotScheduleAssociationParseId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := conn.DescribeSnapshotSchedules(&redshift.DescribeSnapshotSchedulesInput{
+			ScheduleIdentifier: aws.String(scheduleIdentifier),
+			ClusterIdentifier:  aws.String(clusterIdentifier),
+		})
+
+		if err == nil {
```

Comment:
We typically prefer to invert this logic so it reads a little easier:

```go
if err != nil {
  return err
}

if resp != nil && len(resp.SnapshotSchedules) > 0 {
  return fmt.Errorf(/*...*/)
}

// no return nil since we want to check all the resources of this type
```